packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -159,6 +185,181 @@ async function createOutgoingPayment(
   }
 }
 
+function validateAccess({
+  access,
+  payment,
+  identifier
+}: {
+  access: GrantAccess
+  payment: OutgoingPayment
+  identifier: string
+}): PaymentAccess | undefined {
+  let paymentInterval: Interval | undefined
+  if (access.interval)
+    paymentInterval = getInterval(access.interval, payment.createdAt)
+  if (
+    (!access.identifier || access.identifier === identifier) &&
+    (!access.interval || paymentInterval !== undefined) &&
+    (!access.limits || validateAccessLimits(payment, access.limits))
+  ) {
+    return { ...access, paymentInterval }
+  }
+}
+
+function validatePaymentAccess({
+  access,
+  payment,
+  identifier
+}: {
+  access: PaymentAccess
+  payment: OutgoingPayment
+  identifier: string
+}): boolean {
+  if (
+    (!access.identifier || access.identifier === identifier) &&
+    (!access.paymentInterval ||
+      (access.paymentInterval.start.toMillis() <= payment.createdAt.getTime() &&
+        payment.createdAt.getTime() < access.paymentInterval.end.toMillis())) &&
+    (!access.limits || validateAccessLimits(payment, access.limits))
+  ) {
+    return true
+  }
+  return false
+}
+
+function validateAccessLimits(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): boolean {
+  if (
+    (!limits.receiver || payment.receiver === limits?.receiver) &&
+    (!limits.description || payment.description === limits?.description) &&
+    (!limits.externalRef || payment.externalRef === limits?.externalRef) &&
+    validateAmountAssets(payment, limits)
+  ) {
+    return true
+  }
+  return false
+}
+
+function validateAmountAssets(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): boolean {
+  if (
+    limits.sendAmount &&
+    (limits.sendAmount.assetCode !== payment.asset.code ||
+      limits.sendAmount.assetScale !== payment.asset.scale)
+  ) {
+    return false
+  }
+  return (
+    !limits.receiveAmount ||
+    (limits.receiveAmount.assetCode === payment.receiveAmount.assetCode &&
+      limits.receiveAmount.assetScale === payment.receiveAmount.assetScale)
+  )
+}
+
+interface PaymentAccess extends GrantAccess {
+  paymentInterval?: Interval
+}
+
+// "payment" is locked by the "deps.knex" transaction.
+async function validateGrant(
+  deps: ServiceDependencies,
+  payment: OutgoingPayment,
+  grant: Grant
+): Promise<boolean> {
+  const grantAccess = grant.getAccess({
+    type: AccessType.OutgoingPayment,
+    action: AccessAction.Create
+  })
```

Comment:
The [auth middleware](https://github.com/interledger/rafiki/blob/main/packages/backend/src/open_payments/auth/middleware.ts#L28) already calls `grant.includesAccess`
```suggestion
  const grantAccess = grant.access[0]
```